library_generation/test/integration_tests.py

```diff
@@ -12,103 +12,84 @@
-from library_generation.utils.utilities import (
-    sh_util as shell_call,
-    run_process_and_print_output,
-)
+from library_generation.utils.utilities import sh_util as shell_call
 
-config_name = "generation_config.yaml"
 script_dir = os.path.dirname(os.path.realpath(__file__))
-# for simplicity, the configuration files should be in a relative directory
-# within config_dir named {repo}/generation_config.yaml, where repo is
-# the name of the repository the target libraries live.
-config_dir = f"{script_dir}/resources/integration"
-golden_dir = f"{config_dir}/golden"
+golden_dir = os.path.join(script_dir, "resources", "integration", "golden")
+repo_root_dir = os.path.join(script_dir, "..", "..")
+build_file = os.path.join(
+    repo_root_dir, ".cloudbuild", "library_generation", "library_generation.Dockerfile"
+)
+image_tag = f"test-image:latest"
 repo_prefix = "https://github.com/googleapis"
 output_dir = shell_call("get_output_folder")
 # this map tells which branch of each repo should we use for our diff tests
 committish_map = {
     "google-cloud-java": "chore/test-hermetic-build",
     "java-bigtable": "chore/test-hermetic-build",
 }
+config_dir = f"{script_dir}/resources/integration"
+config_name = "generation_config.yaml"
+monorepo_baseline_commit = "a17d4caf184b050d50cacf2b0d579ce72c31ce74"
+split_repo_baseline_commit = "679060c64136e85b52838f53cfe612ce51e60d1d"
 
 
 class IntegrationTest(unittest.TestCase):
-    def test_get_commit_message_success(self):
-        repo_url = "https://github.com/googleapis/googleapis.git"
-        config_files = self.__get_config_files(config_dir)
-        monorepo_baseline_commit = "a17d4caf184b050d50cacf2b0d579ce72c31ce74"
-        split_repo_baseline_commit = "679060c64136e85b52838f53cfe612ce51e60d1d"
-        for repo, config_file in config_files:
-            baseline_commit = (
-                monorepo_baseline_commit
-                if repo == "google-cloud-java"
-                else split_repo_baseline_commit
-            )
-            description = generate_pr_descriptions(
-                generation_config_yaml=config_file,
-                repo_url=repo_url,
-                baseline_commit=baseline_commit,
-            )
-            description_file = f"{config_dir}/{repo}/pr-description.txt"
-            if os.path.isfile(f"{description_file}"):
-                os.remove(f"{description_file}")
-            with open(f"{description_file}", "w+") as f:
-                f.write(description)
-            self.assertTrue(
-                cmp(
-                    f"{config_dir}/{repo}/pr-description-golden.txt",
-                    f"{description_file}",
-                ),
-                "The generated PR description does not match the expected golden file",
-            )
-            os.remove(f"{description_file}")
+    def test_entry_point_running_in_container(self):
+        self.__build_image(docker_file=build_file, tag=image_tag, cwd=repo_root_dir)
```

Comment:
Unless we want to expose `tag` as a parameter of `IntegrationTest` in the future, I don't think we need to pass around `tag` in this script, we can always use `image_tag`.

Reply:
Done.